### PR TITLE
Add dateOfBirth property to user and its requested pipes

### PR DESCRIPTION
**Pull request by mgandore**

### CODE REVIEW

File: component-interaction/src/app/isAdultUser.pipe.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+	name: "isAdult"
+})
+
+export class isAdultUserPipe implements PipeTransform {
```
Comment (danielursu) [inline component-interaction/src/app/isAdultUser.pipe.ts]:
```suggestion
export class IsAdultUserPipe implements PipeTransform {

```

File: component-interaction/src/app/isAdultUser.pipe.ts [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+	name: "isAdult"
+})
+
```
Comment (danielursu) [inline component-interaction/src/app/isAdultUser.pipe.ts]:
```suggestion
```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 3]
```diff
@@ -63,12 +67,24 @@ export class UserListComponent implements OnInit {
...
 
+	public getRandomNumForDate(input?: string): number {
+		if (input === "year") {
+			return Math.floor(Math.random() * 100) + 1900;
+		} else if (input === "month") {
+			return Math.floor(Math.random() * 12) + 1;
+		}
+		return Math.floor(Math.random() * 31) + 1;
+
+
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 4]
```diff
@@ -63,12 +67,24 @@ export class UserListComponent implements OnInit {
 		return randomText;
 	}
 
+	public getRandomNumForDate(input?: string): number {
+		if (input === "year") {
+			return Math.floor(Math.random() * 100) + 1900;
+		} else if (input === "month") {
+			return Math.floor(Math.random() * 12) + 1;
+		}
+		return Math.floor(Math.random() * 31) + 1;
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]: use switch + InputDateEnum

File: component-interaction/src/app/userAge.pipe.ts [hunk 5]
```diff
@@ -0,0 +1,10 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+	name: "getAge"
+})
+export class UserAgePipe implements PipeTransform {
+	public transform(dateOfBirth: any, ...args: any[]): number {
+		return 2022 - dateOfBirth?.getFullYear();
```
Comment (danielursu) [inline component-interaction/src/app/userAge.pipe.ts]:
```suggestion
		return 2022 - dateOfBirth?.getFullYear();

```
will this snippet work next year? :))

**Review by danielursu — CHANGES_REQUESTED** [review]

File: component-interaction/src/app/userAge.pipe.ts (hunk 5 above)
Comment (mgandore) [reply component-interaction/src/app/userAge.pipe.ts]: 😭😭😭🤦‍♂️

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 6]
```diff
@@ -63,12 +69,32 @@ export class UserListComponent implements OnInit {
 		return randomText;
 	}
 
+	public getRandomNumForDate(input: InputDateEnum): number {
+		let output: number = 0;
+		switch (input) {
+			case "year":
+				output = Math.floor(Math.random() * 100) + 1900;
+				break;
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]:
```suggestion
		switch (input) {
			case InputDateEnum.YEAR:
				return Math.floor(Math.random() * 100) + 1900;

```

File: component-interaction/src/app/user-list/user-list.component.ts [hunk 7]
```diff
@@ -63,12 +69,32 @@ export class UserListComponent implements OnInit {
...
+				output = Math.floor(Math.random() * 100) + 1900;
+				break;
+			case "month":
+				output = Math.floor(Math.random() * 12) + 1;
+				break;
+			case "day":
+				output = Math.floor(Math.random() * 31) + 1;
+				break;
+		}
+		return output;
```
Comment (danielursu) [inline component-interaction/src/app/user-list/user-list.component.ts]: same here

Comment (danielursu) [reply component-interaction/src/app/user-list/user-list.component.ts]: also you can get rid of output variable and return 0 after switch

File: component-interaction/src/app/userAge.pipe.ts [hunk 8]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
+
+@Pipe({
+	name: "getAge"
+})
+export class UserAgePipe implements PipeTransform {
+	public transform(dateOfBirth: any, ...args: any[]): number {
+		let currentDate = new Date();
+		return currentDate.getFullYear() - dateOfBirth.getFullYear();
```
Comment (danielursu) [inline component-interaction/src/app/userAge.pipe.ts]:
```suggestion
		return new Date().getFullYear() - dateOfBirth.getFullYear();

```

File: component-interaction/src/app/app.module.ts [hunk 9]
```diff
@@ -2,16 +2,20 @@ import { NgModule } from '@angular/core';
 import { BrowserModule } from '@angular/platform-browser';
 
 import { AppComponent } from './app.component';
+import { IsAdultUserPipe } from './isAdultUser.pipe';
 import { SortPipe } from './sort.pipe';
 import { UserDetailsComponent } from './user-details/user-details.component';
 import { UserListComponent } from './user-list/user-list.component';
+import { UserAgePipe } from './userAge.pipe';
```
Comment (danielursu) [inline component-interaction/src/app/app.module.ts]: double qoutes

File: component-interaction/src/app/inputDate.enum.ts [hunk 10]
```diff
@@ -0,0 +1,5 @@
+export enum InputDateEnum {
+	year = "year",
+	month = "month",
+	day = "day"
```
Comment (danielursu) [inline component-interaction/src/app/inputDate.enum.ts]:
```suggestion
	YEAR = "year",
	MONTH = "month",
	DAY = "day"

```

File: component-interaction/src/app/isAdultUser.pipe.ts [hunk 11]
```diff
@@ -0,0 +1,10 @@
+import { Pipe, PipeTransform } from "@angular/core";
```
Comment (danielursu) [inline component-interaction/src/app/isAdultUser.pipe.ts]: is-adult-user.pipe.ts

File: component-interaction/src/app/userAge.pipe.ts [hunk 12]
```diff
@@ -0,0 +1,11 @@
+import { Pipe, PipeTransform } from "@angular/core";
```
Comment (danielursu) [inline component-interaction/src/app/userAge.pipe.ts]: user-age.pipe.ts

**Review by danielursu — CHANGES_REQUESTED** [review]

**Review by danielursu — APPROVED** [review]

**Pull request merged**